buildSrc/src/main/java/com/google/firebase/gradle/plugins/ProjectUtils.kt

```diff
@@ -63,6 +64,14 @@ fun <T : Task, R : Task> T.dependsOnAndMustRunAfter(otherTask: R) {
     dependsOn(otherTask)
 }
 
+/**
+ * Utility method to call [Task.mustRunAfter] and [Task.dependsOn] on the specified task
+ */
+fun <T : Task, R : Task> T.dependsOnAndMustRunAfter(otherTask: Provider<R>) {
+    mustRunAfter(otherTask)
+    dependsOn(otherTask)
```

Comment:
doesn't `dependsOn` imply `mustRunAfter` though?

Reply:
You would think so, but it doesn't. Sometimes it will run after it, sometimes it will run alongside it. Depending on how gradle decides to optimize things.

Reply:
I see, in that case inputs/outputs seems like the preferred way to model deps iiuc

Reply:
> I see, in that case inputs/outputs seems like the preferred way to model deps iiuc

Hmm, I was having some issues before regarding this- but I'm unable to replicate them anymore. It was like wiring a dependency didn't implicitly force the depended task to be up-to-date. I'll revert this for now, and we can come back to it should the issue pop back up. Could've just been some dependency being cached incorrectly- which we'll wanna find if that's the case.